models/mongo/accounts.go

```diff
@@ -1,3 +1,144 @@
+		return status.Errorf(codes.Internal, "could not create account")
+	}
+	account := account{ID: id.String(), Email: *payload.Email, Name: *payload.Name, Hash: payload.Hash}
+
+	_, err = models.AccountsDatabase.Collection("accounts").InsertOne(ctx, account)
+	if err != nil {
+		srv.logger.Errorw("failed to insert account in db", "error", err.Error(), "email", account.Email)
+		return status.Errorf(codes.Internal, "could not create account")
+	}
+	return nil
+}
+
+func (srv *accountsRepository) Get(ctx context.Context, filter *models.OneAccountFilter) (*models.Account, error) {
+	var account account
+
+	query := buildQuery(filter)
+	err := models.AccountsDatabase.Collection("accounts").FindOne(ctx, query).Decode(&account)
+
+	fmt.Println("account = ", account)
+
+	if err != nil {
+		if errors.Is(err, mongo.ErrNoDocuments) {
+			return nil, status.Errorf(codes.NotFound, "account not found")
+		}
+		srv.logger.Errorw("unable to query accounts", "error", err.Error())
+		return nil, status.Errorf(codes.InvalidArgument, err.Error())
+	}
+
+	uuid, err := uuid.Parse(account.ID)
+	if err != nil {
+		srv.logger.Errorw("failed to convert uuid from string", "error", err.Error())
+		return nil, status.Errorf(codes.Internal, "could not get account")
+	}
+
+	return &models.Account{ID: uuid, Name: account.Name, Email: account.Email, Hash: account.Hash}, nil
+}
+
+func (srv *accountsRepository) Delete(ctx context.Context, filter *models.OneAccountFilter) error {
+
+	query := buildQuery(filter)
+
+	delete, err := models.AccountsDatabase.Collection("accounts").DeleteOne(ctx, query)
+
+	if err != nil {
+		srv.logger.Errorw("delete account db query failed", "error", err.Error())
+		return status.Errorf(codes.Internal, "could not delete account")
+	}
+	if delete.DeletedCount == 0 {
+		srv.logger.Errorw("delete account db query matched none", "user_id", filter.ID)
+		return status.Errorf(codes.Internal, "could not delete account")
+	}
+	return nil
+}
+
+func (srv *accountsRepository) Update(ctx context.Context, filter *models.OneAccountFilter, account *models.AccountPayload) error {
+
+	query := buildQuery(filter)
+	update, err := models.AccountsDatabase.Collection("accounts").UpdateOne(ctx, query, bson.D{{Key: "$set", Value: &account}})
+	if err != nil {
+		srv.logger.Errorw("failed to convert object id from hex", "error", err.Error())
+		return status.Errorf(codes.InvalidArgument, err.Error())
+	}
+	if update.MatchedCount == 0 {
+		srv.logger.Errorw("update account db query matched none", "user_id", filter.ID.String())
+		return status.Errorf(codes.Internal, "could not update account")
+	}
+	return nil
+}
+
+func (srv *accountsRepository) List(ctx context.Context) (*[]models.Account, error) {
+	var accounts []account
+	cursors, err := models.AccountsDatabase.Collection("accounts").Find(ctx, bson.D{})
+	if err != nil {
+		srv.logger.Errorw("failed get documents ", "error", err.Error())
```

Comment:
I would replace this with `"failed to fetch accounts from mongo"`. But I guess we'll do that next sprint because we will most likely implement some error handling standard and libraries to return more accurate errors and to reduce typing.